code/controllers/subsystems/zcopy.dm

```diff
@@ -127,28 +142,35 @@ SUBSYSTEM_DEF(zcopy)
 			T.z_eventually_space = TRUE
 			t_target = SPACE_PLANE
 
-		if (!(T.z_flags & ZM_MIMIC_OVERWRITE))
-			// Some openturfs have icons, so we can't overwrite their appearance.
-			if (!T.below.bound_overlay)
-				T.below.bound_overlay = new(T)
-			var/atom/movable/openspace/turf_overlay/TO = T.below.bound_overlay
-			TO.appearance = T.below
-			TO.name = T.name
-			TO.gender = T.gender	// Need to grab this too so PLURAL works properly in examine.
-			TO.opacity = FALSE
-			TO.plane = t_target
-		else
-			// This openturf doesn't care about its icon, so we can just overwrite it.
-			if (T.below.bound_overlay)
-				QDEL_NULL(T.below.bound_overlay)
-			T.appearance = T.below
-			T.name = initial(T.name)
-			T.desc = initial(T.desc)
-			T.gender = NEUTER
-			T.opacity = FALSE
-			T.plane = t_target
-
-		T.queue_ao()
+		if (T.below.z_flags & ZM_FIX_BIGTURF)
+			T.z_flags |= ZM_FIX_BIGTURF	// this flag is infectious
+
+		// There's no point creating TOs for space, it'll draw under the Z-turf anyways.
+		if (!T.below.z_eventually_space || (T.z_flags & ZM_MIMIC_OVERWRITE))
+			if (T.z_flags & ZM_MIMIC_OVERWRITE)
+				// This openturf doesn't care about its icon, so we can just overwrite it.
+				if (T.below.bound_overlay)
+					QDEL_NULL(T.below.bound_overlay)
+				T.appearance = T.below
+				T.name = initial(T.name)
+				T.desc = initial(T.desc)
+				T.gender = initial(T.gender)
+				T.opacity = FALSE
+				T.plane = t_target
+			else
+				// Some openturfs have icons, so we can't overwrite their appearance.
+				if (!T.below.bound_overlay)
+					T.below.bound_overlay = new(T)
+				var/atom/movable/openspace/turf_overlay/TO = T.below.bound_overlay
+				TO.appearance = T.below
+				TO.name = T.name
+				TO.gender = T.gender	// Need to grab this too so PLURAL works properly in examine.
+				TO.opacity = FALSE
+				TO.plane = t_target
+		else if (T.below.bound_overlay)
+			QDEL_NULL(T.below.bound_overlay)
```

Comment:
Maybe cleaner to do
```
if(T.z_flags & ZM_MIMIC_OVERWRITE)
    // This openturf doesn't care about its icon, so we can just overwrite it.
    etc
else if(T.below.z_eventually_space)
    QDEL_NULL(T.below.bound_overlay)
else
    // Some openturfs have icons, so we can't overwrite their appearance.
    etc
```
though I could be misreading the logic branching here.

Reply:
That should work too; this is structured like this as the code it was extracted from had another branch above the bound_overlay check.